src/lib/ext-popup.js

```diff
@@ -44,46 +45,36 @@ export default class Popup {
       : this.renderMonitorStoppedUI();
   }
 
-  handleViewActivityLog() {
-    openActivityLogPage();
+  async handleViewActivityLog() {
+    await openActivityLogPage();
     window.close();
   }
 
   async handleEvent(event) {
     try {
-      if (event.type === 'click') {
-        switch (event.target) {
-          case this.startMonitorAllBtn:
-            await this.handleMonitor(startMonitor);
-            break;
-          case this.stopMonitorAllBtn:
-            await this.handleMonitor(stopMonitor);
-            break;
-          case this.openActivityLogBtn:
-            this.handleViewActivityLog();
-            break;
-          default:
-            throw new Error(
-              'wrong event target id found: ' + JSON.stringify(event.target.id)
-            );
-        }
-      } else {
-        throw new Error(
-          'wrong event type found: ' + JSON.stringify(event.type)
-        );
+      switch (event.target) {
+        case this.startMonitorAllBtn:
+          await this.handleMonitor(startMonitor);
+          break;
+        case this.stopMonitorAllBtn:
+          await this.handleMonitor(stopMonitor);
+          break;
+        case this.openActivityLogBtn:
+          await this.handleViewActivityLog();
+          break;
+        default:
+          throw new Error(
+            'wrong event target id found: ' + JSON.stringify(event.target.id)
+          );
       }
     } catch (error) {
       this.renderErrorMsg(error.message);
     }
   }
 
   async handleMonitor(monitorFunc) {
-    try {
-      await monitorFunc();
-      this.render();
-    } catch (error) {
-      this.renderErrorMsg(error.message);
-    }
+    await monitorFunc();
```

Comment:
Why did we have try-catch before, and how do we know that removing try-catch is fine?